Add support for DUnit

Zalastax opened the pull request:
My code works, but DUnit is having problems on one compiler. I don't DUnit is worth losing support for a platform, but I think we could try to find out what's causing the error and file a file a bug report/pull request.

Zalastax committed:
Add configuration for library and unittest.

This allows unit tests to depend on dunit without forcing our users to depend on dunit.

Zalastax committed:
Import dunit.toolkit in unit test version.

Zalastax committed:
Use assertions from dunit.toolkit

Zalastax committed:
Remove unused methods jsonEquals

Using jsonEquals hide valuable information from the failure output

BlackEdder commented:
So does dub test automatically use the unittest configuration? I didn't know dub would automatically do that. That does seem to solve my objection in #23.

I guess losing support for an _old_ ldc2 version is not that big a deal. And we would only lose testing support I guess. It should still be usable, just not tested.

Zalastax committed:
Removed testing for ldc 0.14.0

Since DUnit doesn't work for ldc 0.14.0 we have to remove testing support for it.

Zalastax committed:
Update unit test types PointUseConstructor and IdAndName to have more realistic usage.

Change fields to be immutable or private with a getter to give them a reason to have their constructor disabled..

Zalastax committed:
Change IdAndName to class.

Zalastax committed:
Replace assert(0) in constructor of IdAndName to a working constructor.

Zalastax committed:
Update README.md

Zalastax commented:
Yeah it seems like dub automatically uses the configuration named unittest. This feature seems to be undocumented, I found out about it in the forums.  

This pull request can be merged now.
